mathics/core/evaluation.py

```diff
@@ -184,11 +184,11 @@ def __init__(
         # ``mathics.builtin.numeric.N``.
         self._preferred_n_method = []
 
-    def parse(self, query):
+    def parse(self, query, src_name: str = ""):
```

Comment:
with this change, we can ask `MathicsSession` to tell the parser that the source is `"<test>"` or whatever other tag that could be useful.